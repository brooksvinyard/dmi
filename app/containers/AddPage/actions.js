/*
 *
 * AddPage actions
 *
 */

import {
  CHANGE_STRING,
  POST_STRINGS,
  POST_STRINGS_SUCCESS,
  POST_STRINGS_ERROR,
} from './constants';

export function changeString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}

export function postString() {
  return {
    type: POST_STRINGS,
  };
}

export function postStringSuccess() {
  return {
    type: POST_STRINGS_SUCCESS,
  };
}

export function postStringError(error) {
  return {
    type: POST_STRINGS_ERROR,
    error,
  };
}
