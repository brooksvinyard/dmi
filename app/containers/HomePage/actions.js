/*
 * Home Actions
 *
 */

import { GET_STRINGS } from './constants';

/**
 * Gets the strings from the database
 *
 * @return {object} An action object with a type of GET_STRINGS
 */
export function getStrings() {
  return {
    type: GET_STRINGS,
  };
}
