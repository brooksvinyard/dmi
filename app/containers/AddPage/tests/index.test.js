/**
 *
 * Tests for AddPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

// import 'jest-dom/extend-expect'; // add some helpful assertions

import { AddPage, mapDispatchToProps } from '../index';
import { changeString, postString } from '../actions';

import configureStore from '../../../configureStore';

describe('<AddPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <AddPage loading={false} error={false} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot after success', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <AddPage loading={false} error={false} postSuccess />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot after error', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <AddPage loading={false} error />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should not call onPostString if string is an empty string', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <AddPage onChangeString={() => {}} onPostString={submitSpy} />
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  it('should not call onPostString if username is null', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <AddPage string="" onChangeString={() => {}} onPostString={submitSpy} />
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(<AddPage dispatch={dispatch} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<AddPage />);
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeString).toBeDefined();
      });

      it('should dispatch changeString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const string = 'Hello World';
        result.onChangeString({ target: { value: string } });
        expect(dispatch).toHaveBeenCalledWith(changeString(string));
      });
    });

    describe('onPostString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onPostString).toBeDefined();
      });

      it('should dispatch postString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onPostString();
        expect(dispatch).toHaveBeenCalledWith(postString());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const e = { preventDefault };
        result.onPostString(e);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
