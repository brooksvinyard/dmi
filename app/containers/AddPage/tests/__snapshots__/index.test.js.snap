// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPage /> Should render and match the snapshot 1`] = `
<div
  class="Paper-bhubno-0 bHtalL"
>
  <h2
    class="H2-egphza-0 ljvrU"
  >
    Add a new String
  </h2>
  <p
    class="P-sc-76wrpw-0 bTCnCC"
  >
    Add a new string below.
  </p>
  <form>
    <input
      class="Input-sc-19txxuq-0 jwLKGP"
      id="string"
      placeholder="Hello World!"
      required=""
      type="text"
      value=""
    />
    <button
      class="Button-sc-1fkoyq4-0 cSUWIY"
    >
      Add String
    </button>
  </form>
</div>
`;

exports[`<AddPage /> should render and match the snapshot 1`] = `
<div
  class="Paper-bhubno-0 bHtalL"
>
  <h2
    class="H2-egphza-0 ljvrU"
  >
    Add a new String
  </h2>
  <p
    class="P-sc-76wrpw-0 bTCnCC"
  >
    Add a new string below.
  </p>
  <form>
    <input
      class="Input-sc-19txxuq-0 jwLKGP"
      id="string"
      placeholder="Hello World!"
      required=""
      type="text"
      value=""
    />
    <button
      class="Button-sc-1fkoyq4-0 cSUWIY"
    >
      Add String
    </button>
  </form>
</div>
`;

exports[`<AddPage /> should render and match the snapshot after error 1`] = `
<div
  class="Paper-bhubno-0 bHtalL"
>
  <h2
    class="H2-egphza-0 ljvrU"
  >
    Add a new String
  </h2>
  <p
    class="P-sc-76wrpw-0 bTCnCC"
  >
    Add a new string below.
  </p>
  <form>
    <input
      class="Input-sc-19txxuq-0 jwLKGP"
      id="string"
      placeholder="Hello World!"
      required=""
      type="text"
      value=""
    />
    <button
      class="Button-sc-1fkoyq4-0 cSUWIY"
    >
      Add String
    </button>
  </form>
  <div
    class="Error-sc-50fjew-0 iNXEva"
  >
    There was an error adding the string to the database.
  </div>
</div>
`;

exports[`<AddPage /> should render and match the snapshot after success 1`] = `
<div
  class="Paper-bhubno-0 bHtalL"
>
  <h2
    class="H2-egphza-0 ljvrU"
  >
    Add a new String
  </h2>
  <p
    class="P-sc-76wrpw-0 bTCnCC"
  >
    Add a new string below.
  </p>
  <form>
    <input
      class="Input-sc-19txxuq-0 jwLKGP"
      id="string"
      placeholder="Hello World!"
      required=""
      type="text"
      value=""
    />
    <button
      class="Button-sc-1fkoyq4-0 cSUWIY"
    >
      Add String
    </button>
  </form>
  <p
    class="P-sc-76wrpw-0 bTCnCC"
  >
    String successfully added to database.
  </p>
</div>
`;
